pymeasure/instruments/rohdeschwarz/fsl.py

```diff
@@ -23,238 +23,19 @@
-        :return: 2d numpy array of the trace data, [[frequency], [amplitude]].
-        """
-        y = np.array(self.values(f"TRAC{n_trace}? TRACE{n_trace}"))
-        x = np.linspace(self.freq_start, self.freq_stop, len(y))
-        return np.array([x, y])
-
-    trace_mode = Instrument.control(
-        "DISP:TRAC:MODE?",
-        "DISP:TRAC:MODE %s",
-        "Trace mode ('WRIT', 'MAXH', 'MINH', 'AVER' or 'VIEW')",
-        validator=strict_discrete_set,
-        values=["WRIT", "MAXH", "MINH", "AVER", "VIEW"],
-    )
-
-    # Markers -----------------------------------------------------------------
-
-    def create_marker(self, num=1, is_delta_marker=False):
-        """
-        Create a marker.
-
-        :param num: The marker number (1-4)
-        :param is_delta_marker: True if the marker is a delta marker, default
-            is False.
-        :return: The marker object.
-        """
-        return self.Marker(self, num, is_delta_marker)
-
-    class Marker:
-        def __init__(self, instrument, num, is_delta_marker):
-            """
-            Marker and Delta Marker class.
-
-            :param instrument: The FSL instrument.
-            :param num: The marker number (1-4)
-            :param is_delta_marker: True if the marker is a delta marker,
-                defaults to False.
-            """
-            self.instrument = instrument
-            self.is_delta_marker = is_delta_marker
-            # Building the marker name for the commands.
-            if self.is_delta_marker:
-                # Smallest delta marker number is 2.
-                self.name = "DELT" + str(max(2, num))
-            else:
-                self.name = "MARK"
-                if num > 1:
-                    # Marker 1 doesn't get a number.
-                    self.name = self.name + str(num)
-
-            self.activate()
-
-        def read(self):
-            return self.instrument.read()
-
-        def write(self, command):
-            self.instrument.write(f"CALC:{self.name}:{command}")
-
-        def ask(self, command):
-            return self.instrument.ask(f"CALC:{self.name}:{command}")
-
-        def values(self, command, **kwargs):
-            """
-            Reads a set of values from the instrument through the adapter,
-            passing on any keyword arguments.
-            """
-            return self.instrument.values(
-                f"CALC:{self.name}:{command}", **kwargs
-            )
-
-        def activate(self):
-            """Activate a marker."""
-            self.write("STAT ON")
-
-        def disable(self):
-            """Disable a marker."""
-            self.write("STAT OFF")
-
-        x = Instrument.control(
-            "X?", "X %s", "Position of marker on the frequency axis in Hz."
+        warn(
+            "Import `FSL` from `pymeasure.instruments.rohdeschwarz.fsl` is deprecated. Import "
+            "`FSL` from `pymeasure.instruments.rohdeschwarz` instead.",
+            FutureWarning,
```

Comment:
Please add a note, since when this is deprecated (the next release, here it is 0.15).
Also, please add a section about this file name change to the `changes.rst file`, such that users will find the file name change in the changelog.

Reply:
I added it now to the changes.rst under "upcoming version" (3f9713c), hope that's as intended!

Reply:
yes, it is the right place, but please open there a new section "deprecated"